Src/FluentAssertions/Equivalency/Node.cs

```diff
@@ -94,25 +91,29 @@ public static INode From<T>(GetSubjectId getSubjectId)
 
     public static INode FromCollectionItem<T>(string index, INode parent)
     {
+        Pathway.GetDescription getDescription = pathAndName => $"{parent.GetSubjectId().Combine(pathAndName)}";
+
         return new Node
         {
             Type = typeof(T),
             ParentType = parent.Type,
-            Name = "[" + index + "]",
-            Path = parent.PathAndName,
+            Subject = new Pathway(parent.Subject, "[" + index + "]", getDescription),
+            Expectation = new Pathway(parent.Expectation, "[" + index + "]", getDescription),
```

Comment:
Suggestion: cache `"[" + index + "]"` in a variable.

Same suggestion for `"[" + key + "]"` in `FromDictionaryItem`